cpp/src/io/parquet/reader_impl.cu

```diff
@@ -1222,6 +1237,17 @@ rmm::device_buffer reader::impl::decompress_page_data(
 
   decompress_check(comp_res, _stream);
 
+  // now copy the uncompressed V2 def and rep level data
+  if (not copy_in.empty()) {
+    host_span<device_span<uint8_t const> const> copy_in_view{copy_in.data(), copy_in.size()};
+    auto const d_copy_in = cudf::detail::make_device_uvector_async(copy_in_view, _stream);
+
+    host_span<device_span<uint8_t> const> copy_out_view(copy_out.data(), copy_out.size());
+    auto const d_copy_out = cudf::detail::make_device_uvector_async(copy_out_view, _stream);
```

Comment:
I think I see why you need to create the spans here. Maybe this is better?
```suggestion
    auto const d_copy_in = cudf::detail::make_device_uvector_async<device_span<uint8_t const> const>(copy_in_view, _stream);
    auto const d_copy_out = cudf::detail::make_device_uvector_async<device_span<uint8_t> const>(copy_out_view, _stream);
```

Reply:
I was just copying what's going on around line 1191.  I'll see what I can do here.

Reply:
Thanks @vuule !  I didn't realize `make_device_uvector_async` was overloaded like that.  Much simpler now 😄

Reply:
Are you refering to code after line 1191? AFAICT, we need to create spans there because of `start_pos`.

Reply:
Ah, I see it.  Ok.